### PR TITLE
Add Electron support

### PR DESCRIPTION
**Pull request by stevenvachon**

### CODE REVIEW

**Comment by stevenvachon** [comment]
@phillipskevin can we get this merged today?

File: resources/electron-preload.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+(function(){
+  'use strict';
+  const {ipcRenderer} = require('electron');
```
Comment (phillipskevin) [inline resources/electron-preload.js]: I don't think destructuring is supported in Node 4. Should this all be written in ES5?

File: resources/electron.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// An Electron sript that simply opens the web page passed as the command line parameter
```
Comment (phillipskevin) [inline resources/electron.js]: `sript` -> `script`

File: test/local.js [hunk 3]
```diff
@@ -58,22 +58,68 @@ describe('Local browser launcher tests', function() {
...
+
+      after(function () {
+        delete process.env.LAUNCHPAD_ELECTRON;
+      });
+
+      it('is detected only due to env settings', function (done) {
+        local(function (error, launcher) {
+          launcher.browsers(function (error, browsers) {
+            assert.ok(!error, 'No error discovering browsers');
+            assert.ok(browsers.length == 1, 'Found Electron browser');
```
Comment (phillipskevin) [inline test/local.js]: These could use `assert.equal`, or at least `===`.

File: resources/electron.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+'use strict';
+const {app, BrowserWindow, ipcMain} = require('electron');
```
Comment (phillipskevin) [inline resources/electron.js]: Same as above... should use ES5.

File: test/local.js [hunk 5]
```diff
@@ -58,22 +58,68 @@ describe('Local browser launcher tests', function() {
...
+
+      after(function () {
+        delete process.env.LAUNCHPAD_PHANTOM;
+      });
+
+      it('is detected only due to env settings', function (done) {
+        local(function (error, launcher) {
+          launcher.browsers(function (error, browsers) {
+            assert.ok(!error, 'No error discovering browsers');
+            assert.ok(browsers.length == 1, 'Found PhantomJS browser');
```
Comment (phillipskevin) [inline test/local.js]: Same as above... `assert.equal`.

File: resources/electron.js (hunk 4 above)
Comment (phillipskevin) [inline resources/electron.js]: Same as above... should use ES5.

File: .travis.yml [hunk 6]
```diff
@@ -1,10 +1,13 @@
...
-  - "sh -e /etc/init.d/xvfb start"
-  - "sudo apt-get update"
-  - "sudo apt-get install chromium-browser"
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get update; fi
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get install chromium-browser; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew cask install chromium; fi
+os:
+  - linux
+  - osx
```
Comment (phillipskevin) [inline .travis.yml]: Any reason you want to test on osx?

File: resources/electron-preload.js (hunk 1 above)
Comment (stevenvachon) [inline resources/electron-preload.js]: This code is only ran in Electron. I don't think anyone builds desktop apps with old Electron versions running on Node 4.

File: resources/electron.js (hunk 4 above)
Comment (stevenvachon) [inline resources/electron.js]: Ran in Electron only.

File: test/local.js (hunk 3 above)
Comment (stevenvachon) [inline test/local.js]: I'd copied this from other tests. Will change.

File: .travis.yml (hunk 6 above)
Comment (stevenvachon) [inline .travis.yml]: The library has code specific to OS X, so it should be tested.

Comment (phillipskevin) [inline .travis.yml]: 👍

File: resources/electron-preload.js (hunk 1 above)
Comment (phillipskevin) [inline resources/electron-preload.js]: Ok, makes sense.

File: resources/electron.js (hunk 4 above)
Comment (phillipskevin) [inline resources/electron.js]: Ok.

**Comment by stevenvachon** [comment]
Should be ready to go now.

**Comment by stevenvachon** [comment]
@phillipskevin merge?

**Comment by stevenvachon** [comment]
@phillipskevin is there something keeping this from being merged?

**Comment by phillipskevin** [comment]
Sorry this slipped my mind. Since this was recently moved to the Bitovi org I didn't have merge rights on it. I will merge now. I'll also publish a new minor release, sound good to you? @stevenvachon

**Comment by stevenvachon** [comment]
@phillipskevin yup, np

**Comment by stevenvachon** [comment]
@phillipskevin when is "now"? 😛

**Comment by phillipskevin** [comment]
[0.7.0](https://github.com/bitovi/launchpad/releases/tag/v0.7.0) is out now.

**Comment by stevenvachon** [comment]
Thanks!

**Pull request merged**